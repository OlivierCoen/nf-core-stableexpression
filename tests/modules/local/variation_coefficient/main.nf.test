nextflow_process {

    name "Test Process VARIATION_COEFFICIENT"
    script "modules/local/variation_coefficient/main.nf"
    process "VARIATION_COEFFICIENT"
    tag "var_coeff"
    tag "module"

    test("Should run without failures") {

        when {

            process {
                """
                ch_csv_files = Channel.fromPath( '$baseDir/tests/input/variation_coefficient/*.csv', checkIfExists: true)
                input[0] = ch_csv_files.collect()
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
