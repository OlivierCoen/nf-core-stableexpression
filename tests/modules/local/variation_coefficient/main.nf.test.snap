{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "variation_coefficients.csv:md5,3b7f865c61386784acf289c526b14e55"
                ],
                "1": [
                    [
                        "VARIATION_COEFFICIENT",
                        "R",
                        "4.3.3 (2024-02-29)"
                    ]
                ],
                "csv": [
                    "variation_coefficients.csv:md5,3b7f865c61386784acf289c526b14e55"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-29T00:31:42.877767395"
    }
}